src/Parsers.jl

```diff
@@ -170,6 +170,62 @@ Options(;
 const OPTIONS = Options(nothing, UInt8(' '), UInt8('\t'), UInt8('"'), UInt8('"'), UInt8('"'), nothing, UInt8('.'), nothing, nothing, nothing, false, false, nothing, false, false, false, nothing)
 const XOPTIONS = Options(missing, UInt8(' '), UInt8('\t'), UInt8('"'), UInt8('"'), UInt8('"'), UInt8(','), UInt8('.'), nothing, nothing, nothing, false, false, nothing, true, false, false, nothing)
 
+abstract type AbstractTypeParseConf{T} end
+abstract type AbstractDateTimeParseConf{T} <: AbstractTypeParseConf{T} end
+abstract type AbstractFloatParseConf{T}    <: AbstractTypeParseConf{T} end
+abstract type AbstractIntParseConf{T}      <: AbstractTypeParseConf{T} end
+abstract type AbstractStringParseConf{T}   <: AbstractTypeParseConf{T} end
+
+struct BoolConf{Bool} <: AbstractTypeParseConf{Bool}
+    refs::Vector{String}
+    trues::Union{Nothing, Vector{PtrLen}}
+    falses::Union{Nothing, Vector{PtrLen}}
+    # TODO: kwargs + validate inputs like Options does
+    BoolConf(::Type{T}=Bool, args...; kwargs...) where {T} = BoolConf{T}(args...; kwargs...)
+    BoolConf{Bool}(options::Options=OPTIONS, refs=options.refs, trues=options.trues, falses=options.falses) = new{Bool}(refs, trues, falses)
+    BoolConf{Bool}(refs, trues, falses) = new{Bool}(refs, trues, falses)
+end
+struct DateTimeConf{T<:Union{DateTime,Date,Time}} <: AbstractDateTimeParseConf{T}
+    dateformat::Union{Nothing, Format}
+    # TODO: kwargs + validate inputs like Options does
+    DateTimeConf(::Type{T}, args...; kwargs...) where {T} = DateTimeConf{T}(args...; kwargs...)
+    DateTimeConf{T}(options::Options=OPTIONS, fmt=options.dateformat) where{T} = new{T}(fmt)
+    DateTimeConf{T}(fmt) where{T} = new{T}(fmt)
+end
+struct FloatConf{T<:Real} <: AbstractFloatParseConf{T}
+    decimal::UInt8
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    FloatConf(::Type{T}, args...; kwargs...) where {T} = FloatConf{T}(args...; kwargs...)
+    FloatConf{T}(options::Options=OPTIONS, decimal=options.decimal, groupmark=options.groupmark) where {T} = new{T}(decimal, groupmark)
+    FloatConf{T}(decimal, groupmark) where {T} = new{T}(decimal, groupmark)
+end
+struct IntConf{T<:Integer} <: AbstractIntParseConf{T}
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    IntConf(::Type{T}, args...; kwargs...) where {T} = IntConf{T}(args...; kwargs...)
+    IntConf{T}(options::Options=OPTIONS, groupmark=options.groupmark) where {T} = new{T}(groupmark)
+    IntConf{T}(groupmark) where {T} = new{T}(groupmark)
+end
+struct StringConf{T<:Union{AbstractString,Char,Symbol}} <: AbstractStringParseConf{T}
+    stripwhitespace::Bool
```

Comment:
We currently rely on `stripwhitespace` for Int/Float parsing, though I don't think we would actually ever have a scenario where people would want to _disable_ that, so we're probably fine just hard-coding `stripwhitespace=true` for numerics.

Reply:
Do you mean hard-coding that in the `xparse` method that touches Floats and Ints? That could work 👍 I was trying to avoid any sort of breaking change in this PR, but this does sound like an unnecessary option for numerics.